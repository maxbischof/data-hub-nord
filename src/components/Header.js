import React from 'react'
import styled from 'styled-components'
import Logo from '../icons/datahubnord_logo.js'
import { useLocation } from 'react-router-dom'

export default function Header() {
  const isRootPath = useLocation().pathname === "/"
  const color = isRootPath ? "white" : "var(--grey)"
  const position = isRootPath ? "absolute" : "static"

  return (
    <StyledHeader color={color} position={position}>
      <a href="/">
        <Logo color={color}/>
        <span>DataHubNord</span>
      </a>
    </StyledHeader>
  )
}

const StyledHeader = styled.header`
  position: ${props => props.position};
  
  span {
    margin: 0 0 0 12px;
  }

  a {
    color: ${props => props.color};
    font-weight: 700;
    padding: 26px 0 30px 22px;
    display: flex;
    align-items: center;
  }

  a:visited {
    color: ${props => props.color};
  }
`

