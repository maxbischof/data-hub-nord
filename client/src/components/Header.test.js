import React from 'react'
import Header from './Header'
import { render } from '@testing-library/react'
import { MemoryRouter } from 'react-router-dom'
import renderer from 'react-test-renderer'

it('renders datahubnord text', () => {
  const { getByText } = render(
    <MemoryRouter initialEntries={['/']}>
      <Header />
    </MemoryRouter>
  )
  expect(getByText('DataHubNord')).toBeInTheDocument()
})

it('renders header with rootpage style ', () => {
  const renderedHeader = renderer
    .create(
      <MemoryRouter initialEntries={['/']}>
        <Header />
      </MemoryRouter>
    )
    .toJSON()

  expect(renderedHeader).toMatchSnapshot()
})

it('renders header with datasetdetails style ', () => {
  const renderedHeader = renderer
    .create(
      <MemoryRouter initialEntries={[`/datensaetze/Arbeitslosigkeit-Kiel-1`]}>
        <Header />
      </MemoryRouter>
    )
    .toJSON()

  expect(renderedHeader).toMatchSnapshot()
})
