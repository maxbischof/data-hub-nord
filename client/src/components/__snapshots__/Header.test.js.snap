// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders header with datasetdetails style  1`] = `
<header
  className="sc-gtssRu bEwGgx"
  color="var(--grey)"
>
  <a
    href="/"
  >
    <svg
      className="sc-bdnylx dmNGwP"
      color="var(--grey)"
    >
      datahubnord_logo.svg
    </svg>
    <span>
      DataHubNord
    </span>
  </a>
</header>
`;

exports[`renders header with rootpage style  1`] = `
<header
  className="sc-gtssRu rMeiG"
  color="white"
>
  <a
    href="/"
  >
    <svg
      className="sc-bdnylx cWfrAZ"
      color="white"
    >
      datahubnord_logo.svg
    </svg>
    <span>
      DataHubNord
    </span>
  </a>
</header>
`;
